clang/test/SemaSYCL/intel-max-global-work-dim.cpp

```diff
@@ -0,0 +1,104 @@
+// RUN: %clang %s -fsyntax-only -fsycl-device-only -DTRIGGER_ERROR -Xclang -verify
+// RUN: %clang %s -fsyntax-only -Xclang -ast-dump -fsycl-device-only | FileCheck %s
+// RUN: %clang_cc1 -fsycl-is-host -fsyntax-only -verify %s
+
+#ifndef __SYCL_DEVICE_ONLY__
+struct FuncObj {
+  [[intelfpga::max_global_work_dim(1)]] // expected-no-diagnostics
+  void operator()() {}
+};
+
+template <typename name, typename Func>
+void kernel(Func kernelFunc) {
+  kernelFunc();
+}
+
+void foo() {
+  kernel<class test_kernel1>(
+      FuncObj());
+}
+
+#else // __SYCL_DEVICE_ONLY__
+
+[[intelfpga::max_global_work_dim(2)]] // expected-warning{{'max_global_work_dim' attribute ignored}}
+void func_ignore() {}
+
+struct FuncObj {
+  [[intelfpga::max_global_work_dim(1)]]
+  void operator()() {}
+};
+
+struct TRIFuncObjGood1 {
+  [[intelfpga::max_global_work_dim(0)]]
+  [[intelfpga::max_work_group_size(1, 1, 1)]]
+  [[cl::reqd_work_group_size(1, 1, 1)]]
```

Comment:
```suggestion
  [[cl::reqd_work_group_size(1, 1, 1)]] // expected-no-diagnostics
```

Reply:
There is expected-warning above and therefore expected-no-diagnostics isn't applicable (it applies to all the compilable code, but not only to a specific string).